ClientTests/TabManagerTests.swift

```diff
@@ -468,9 +470,9 @@ class TabManagerTests: XCTestCase {
         // We add 2 tabs. Then a private one before adding another normal tab and selecting the first.
         // Make sure that when the last one is deleted we dont switch to the private tab
         let deleted = manager.addTab()
-        let newSelected = manager.addTab()
-        manager.addTab(isPrivate: true)
-        manager.addTab()
+        let newSelected = manager.addTab(self.request)
+        manager.addTab(self.request, isPrivate: true)
+        manager.addTab(self.request)
```

Comment:
Can we please add two tests:
1. add two new tabs and check that tab count does not increase
2. restore a session with a new tab and add another one, with same check